### PR TITLE
Update whisk.invoke() and whisk.trigger() doc

### LINKED ISSUE #1283: whisk.invoke() doc needs clarification - blocking
User @glikson pointed out that the documentation and example in `reference.md` doesn't clearly explain the `blocking` parameter. I believe he took the current explanation to mean that the `invoke()` function itself would be synchronous if `blocking: true` which is definitely not the case.

This confusion has come up several times, and the documentation could do better to make it clear what `blocking` really means.

Also, the example code snippet doesn't return the Promise returned from `whisk.invoke()`

### PR DESCRIPTION
**Pull request by bjustin-ibm**
Resolves #1283 

Clarify what `blocking` means
Remove references to `next` callback

### CODE REVIEW

File: docs/reference.md [hunk 1]
```diff
@@ -68,8 +68,8 @@ Additionally, there is no guarantee that actions will execute atomically. Two ac
 
 When an invocation request is received, the system records the request and dispatches an activation.
 
-The system returns an activation ID (in the case of a nonblocking invocation) to confirm that the invocation was received. 
-Notice that if there's a network failure or other failure which intervenes before you receive an HTTP response, it is possible 
+The system returns an activation ID (in the case of a nonblocking invocation) to confirm that the invocation was received.
+Notice that if there's a network failure or other failure which intervenes before you receive an HTTP response, it is possible
```
Comment (bjustin-ibm) [inline docs/reference.md]: Just EOL changes that Atom insisted on making...

Comment (csantanapr) [inline docs/reference.md]: it's always good to clean those trailing spaces

File: docs/reference.md [hunk 2]
```diff
@@ -196,67 +196,51 @@ Notice that regardless of whether an activation is synchronous or asynchronous,
...
 ### Additional SDK methods
 
-The `whisk.invoke()` function invokes another action. It takes as an argument a dictionary that defines the following parameters:
+The `whisk.invoke()` function invokes another action and returns a Promise for the resulting activation. It takes as an argument a dictionary that defines the following parameters:
 
 - *name*: The fully qualified name of the action to invoke,
 - *parameters*: A JSON object that represents the input to the invoked action. If omitted, defaults to an empty object.
-- *apiKey*: The authorization key with which to invoke the action. Defaults to `whisk.getAuthKey()`. 
-- *blocking*: Whether the action should be invoked in blocking or non-blocking mode. Defaults to `false`, indicating a non-blocking invocation.
-- *next*: An optional callback function to be executed when the invocation completes. If next is not supplied, `whisk.invoke()` returns a promise.
```
Comment (csantanapr) [inline docs/reference.md]: Good next is gone. no more explaining it

File: docs/reference.md [hunk 3]
```diff
@@ -196,67 +196,51 @@ Notice that regardless of whether an activation is synchronous or asynchronous,
...
-  - When defined, `activation` is a dictionary with the following fields:
+`whisk.invoke()` returns a Promise. In order to make the OpenWhisk system wait for the invoke to finish, you must return this Promise from your action's `main` function.
+- If the invocation fails, the promise will reject with an object describing the failed invocation. It will potentially have two fields:
+  - *error*: An object describing the error - usually a string.
+  - *activation*: An optional dictionary that may or may not be present depending on the nature of the invocation failure. If present, it will have the following fields:
     - *activationId*: The activation ID:
     - *result*: If the action was invoked in blocking mode: The action result as a JSON object, else `undefined`.
+- If the invocation succeeds, the promise will resolve with a dictionary describing the activation with fields *activationId* and *result* as described above.
+
+Below is an example of a blocking invocation that utilizes the returned promise:
```
Comment (csantanapr) [inline docs/reference.md]: Yes like that you added an example here to explain.

File: docs/reference.md [hunk 4]
```diff
@@ -196,67 +196,51 @@ Notice that regardless of whether an activation is synchronous or asynchronous,
...
-The signature for `next` is `function(error, activation)`, where:
-
-- `error` is `false` if the firing succeeded, and a *truthy* value  if it failed, usually a string that describes the error.
-- On errors, `activation` might be undefined, depending on the failure mode.
-- When defined, `activation` is a dictionary with an `activationId` field that contains the activation ID.
 
-  If `next` is not provided, then `whisk.trigger()` returns a promise.
-  - If the trigger fails, the promise will reject with an object describing the error.
-  - If the trigger succeeds, the promise will resolve with a dictionary with an `activationId` field containing the activation ID.
+`whisk.trigger()` returns a Promise. If you requre the OpenWhisk system wait for the trigger to complete, you should return this Promise from your action's `main` function.
```
Comment (csantanapr) [inline docs/reference.md]: typo `requre`

Comment (csantanapr) [inline docs/reference.md]: missing the `to` in  wait for the

Comment (bjustin-ibm) [inline docs/reference.md]: Fixed both of these...

**Comment by csantanapr** [comment]
LGTM

**Comment by csantanapr** [comment]
rebase and merged ff823d5bd33b9c603a2993d70b9004f013e9ff5c

**Pull request closed without merging**